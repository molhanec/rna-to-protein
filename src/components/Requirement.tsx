import React, { ReactNode } from 'react'
import { ListGroupItem } from 'reactstrap'

interface Props {
  invalid: boolean
  children: ReactNode
}

const Requirement = ({ invalid, children }: Props) => (
  <ListGroupItem color={invalid ? 'danger' : ''} >
    {children}
  </ListGroupItem>
)

export default Requirement
