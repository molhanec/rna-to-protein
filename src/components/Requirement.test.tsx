import React from 'react'
import { render } from '@testing-library/react'
import Requirement from './Requirement'

describe('<Requirement />', () => {
  it('matches snapshot', () => {
    const { container } = render(<Requirement invalid={true}>Msg</Requirement>)
    expect(container).toMatchSnapshot()
  })
})
