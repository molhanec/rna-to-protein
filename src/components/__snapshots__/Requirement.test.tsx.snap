// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Requirement /> matches snapshot 1`] = `
<div>
  <li
    class="list-group-item-red list-group-item"
  >
    Msg
  </li>
</div>
`;
